taskcluster/ci/generate-baseline-profiles/kind.yml

```diff
@@ -0,0 +1,56 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - android_taskgraph.transforms.gradle_optimization:transforms
+    - taskgraph.transforms.job:transforms
+    - taskgraph.transforms.task:transforms
+
+kind-dependencies:
+    - toolchain
+    - external-gradle-dependencies
+
+task-defaults:
+    attributes:
+        retrigger: true
+    fetches:
+        toolchain:
+            - android-sdk-linux
+        external-gradle-dependencies:
+            - external-gradle-dependencies.tar.xz
+    optimization:
+        skip-unless-changed: [] # Paths are dynamically added by transforms
+    run-on-tasks-for: []
+    run:
+        using: gradlew
+        use-caches: false
+    treeherder:
+        kind: test
+    worker-type: b-android-large
+    worker:
+        docker-image: {in-tree: base}
+        max-run-time: 7200
+
+tasks:
+    generate-baseline-profile:
+        description: 'Generate baseline profile for fenix.'
+        dependencies:
+            external-gradle-dependencies: external-gradle-dependencies-fenix
+        attributes:
+            build-type: nightly
+            shipping-product: fenix
+        run:
+            pre-gradlew:
+                - ["cd", "fenix"]
+                - ['java', '-version']
+            gradlew: ['generateBaselineProfile', 'copyBaselineProfile']
+        run-on-tasks-for: []
+        include-shippable-secrets: true
+        include-nightly-version: true
+        source-project-name: "fenix"
```

Comment:
So the [decision task error](https://firefox-ci-tc.services.mozilla.com/tasks/T_t3NstJQv2pm9wwx7pt-g/runs/0/logs/public/logs/live.log) you're getting is caused by the presence of these keys.

These tasks run through each of the transform files defined above in order. Some transforms also define schemas that the task most conform to at the start of the transformation process. In this case, the schema error is coming out of the `taskgraph.transforms.job` transforms.

If we look over at [this schema in the Taskgraph repo](https://github.com/taskcluster/taskgraph/blob/526f4d82906eef73ee69505c406c03f5f35de973/src/taskgraph/transforms/job/__init__.py#L31), we can see that none of these three keys are allowed there.

I suspect you copy/pasted these from the `build-apk` kind. Those tasks work because they apply the `build_apk` transforms. If we inspect those transforms, we'll see they actually [pop these keys off of the task](https://github.com/mozilla-mobile/firefox-android/blob/f68469b548ede5dce657e3e1be337596597610e9/taskcluster/android_taskgraph/transforms/build_apk.py#L49), so that by the time those tasks need to satisfy the `job_description_schema`, the offending keys are no longer present.

So if this task doesn't need these keys, you can just delete them and it should work. Otherwise you'll either need to also run them through the `build_apk` transforms (which likely doesn't make sense here), or you'll have to write your own transform file that processes these values and removes them from the task definition.

Let me know if you have any questions!

Reply:
Thank you for the explanation and for sharing all the docs, offline as well! 🙌 Will open this PR for review, would really appreciate your review.